### PR TITLE
chore(j-s): Slack event when subpoena is serviced

### PR DESCRIPTION
**Pull request by unakb**
[Slack event þegar fyrirkall er birt](https://app.asana.com/0/1199153462262248/1208626990151897)

## What

Send a slack event when a subpoena's service status changes

## Why

So we can see changes to subpoena service statuses

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Introduced a new case event type for improved event logging.
	- Added a method to retrieve service status text for subpoenas.
	- Integrated event posting capabilities in the Subpoena service for tracking service status changes.

- **Bug Fixes**
	- Enhanced error handling for unexpected affected rows during updates.

- **Chores**
	- Updated module dependencies to include the EventModule for enhanced functionality.
	- Simplified guard logic in the Limited Access Subpoena Controller to allow optional subpoena existence.

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/subpoena/subpoena.service.ts [hunk 1]
```diff
@@ -239,6 +241,19 @@ export class SubpoenaService {
...
+
+      subpoena.case
+    ) {
+      this.eventService.postEvent(
+        'SUBPOENA_SERVICE_STATUS',
+        subpoena.case,
+        false,
+        { Staða: Subpoena.serviceStatusText(update.serviceStatus) },
+      )
+    }
```
[the remark this answers is not in the document]
Comment (unakb) [reply apps/judicial-system/backend/src/app/modules/subpoena/subpoena.service.ts]: This is already handled in the event.service

File: apps/judicial-system/backend/src/app/modules/subpoena/models/subpoena.model.ts [hunk 2]
```diff
@@ -21,6 +21,20 @@ import { Defendant } from '../../defendant/models/defendant.model'
...
+      ? 'Birt fyrir verjanda'
+      : serviceStatus === ServiceStatus.ELECTRONICALLY
+      ? 'Birt rafrænt'
+      : serviceStatus === ServiceStatus.IN_PERSON
+      ? 'Birt persónulega'
+      : serviceStatus === ServiceStatus.FAILED
+      ? 'Árangurslaus birting'
+      : serviceStatus === ServiceStatus.EXPIRED
+      ? 'Rann út á tíma'
+      : 'Í birtingarferli' // This should never happen
```
Comment (oddsson) [inline apps/judicial-system/backend/src/app/modules/subpoena/models/subpoena.model.ts:35]: Should the comment be on line 34 instead?

**Review by oddsson — APPROVED** [review]
Nice 🙌

File: apps/judicial-system/backend/src/app/modules/subpoena/models/subpoena.model.ts (hunk 2 above)
Comment (unakb) [reply apps/judicial-system/backend/src/app/modules/subpoena/models/subpoena.model.ts:35]: Technically yes according to the current logic but I'm guessing we will start using EXPIRED at some point, the comment is mostly to point out that there aren't any more options for the service status currently

**Pull request merged**